system/atomic_host.py

```diff
@@ -0,0 +1,128 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public licenses
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION='''
+---
+module: atomic_host
+short_description: Manage the atomic host platform
+description:
+    - Manage the atomic host platform
+    - Reboot option for states ```rollback```, ```upgrade``` and ```deploy``` are not supported, the playbook should do explicit reboot
+version_added: "2.2"
+author: "Saravanan KR @krsacme"
+notes:
+    - Host should be an atomic platform (verified by existence of '/run/ostree-booted' file)
+requirements:
+  - atomic
+options:
+    state:
+        description:
+          - The state of the atomic host to be in
+        required: true
+        choices: ["rollback", "upgrade", "rebase", "deploy"]
+        default: null
+    refspec:
+        description:
+          - The refspec to rebase the atomic host to. Applicable only for ```rebase``` state.
+        required: false
+        default: null
+    revision:
+        description:
+          - Checksum or revision to deploy. Applicable only for ```revision``` state.
+        required: false
+        default: null
+'''
+
+EXAMPLES = '''
+
+# Upgrade the atomic host platform (atomic host upgrade)
+- atomic_host: state=upgrade
+
+'''
+
+RETURN = '''
+msg:
+    description: The command standard output
+    returned: always
+    type: string
+    sample: 'No upgrades available'
+'''
+
+import os
+import sys
+
+def core(module):
+    state = module.params['state']
+
+    args = ['atomic', 'host']
+    args.append(state)
+
+    refspec = module.params['refspec']
+    if state == 'rebase':
+        if refspec:
+            args.append(refspec)
+        else:
+            module.fail_json(msg="'refspec' is mandatory to rebase the atomic host")
+    elif refspec:
+        module.fail_json(msg="'refspec' is applicable only for state 'rebase'")
+
+    revision = module.params['revision']
+    if state == 'deploy':
+        if revision:
+            args.append(revision)
+        else:
+            module.fail_json(msg="'revision' is mandatory to deploy the atomic host")
+    elif revision:
+        module.fail_json(msg="'revision' is applicable only for state 'deploy'")
+
+    out = {}
+    err = {}
+    rc = 0
+
+    rc, out, err = module.run_command(args, check_rc=False)
+
+    if rc == 77:
+        module.exit_json(msg="No upgrades available")
+    elif rc != 0:
+        module.fail_json(rc=rc, msg=err)
+    else:
+        module.exit_json(msg=out, changed=True)
```

Comment:
this always returns changed  unless it fails, as explained in the other plugin, this is not what users expect